scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,249 @@
+        fi;;
+    (--replace-localhost|-r)
+        if (( $# > 1 ));
+        then { arg_ip_addr="$2"; shift; }
+        else { echo "$0: missing ip address" >&2; exit 1; }
+        fi;;
+    (--merge|-m)
+        arg_merge=true;;
+    (--output|-o)
+        if (( $# > 1 ));
+        then { arg_output="$2"; shift; }
+        else { echo "$0: missing output filename" >&2; exit 1; }
+        fi;;
+    (--silent|-s)
+        arg_verbose=false;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { >&2 echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/KUBECONFIG="$in_KUBECONFIG" kubectl/generated/KUBECONFIG="$in_KUBECONFIG" kubectl_config/
+echov "Determining the list of kubeconfigs to search:"
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    in_KUBECONFIG="$arg_kubeconfig"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$arg_kubeconfig"
+    fi
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    in_KUBECONFIG="$KUBECONFIG"
+    if [[ "$arg_output" == "" ]] ; then
+        if [[ "$KUBECONFIG" == *":"* ]] ; then
+            # hard to decide may be out_kubeconfig="$HOME/.kube/config"
+            >&2 echo "ERROR: please specify an output kubeconfig filename with \"-o <filename>\"!"
+            exit 1
+        else
+            out_kubeconfig="$KUBECONFIG"
+        fi
+    fi
+else
+    in_KUBECONFIG="$HOME/.kube/config"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$HOME/.kube/config"
+    else
+        out_kubeconfig="$arg_output"
+    fi
+fi
+echov "- input kubeconfigs: \"$in_KUBECONFIG\""
+echov "- output kubeconfig: \"$out_kubeconfig\""
+
+
+# Determine the list of contexts to search
+echov "Determining the list of contexts to search:"
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(KUBECONFIG="$in_KUBECONFIG" kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- found context \"${context}\""
+done
+
+
+echov "Getting kubeconfigs of KubeFlex Control Planes:"
+n=0
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- searching context \"${context}\""
+    cps=($(KUBECONFIG="$in_KUBECONFIG" kubectl --context "${context}" get controlplanes -no-headers -o name 2> /dev/null || true))
+    for i in "${!cps[@]}" ; do # for all control planes in context ${context}
+        name=${cps[i]##*/}
+        if [[ "$arg_names" != "" && ",$arg_names," != *",$name,"* ]] ; then
+            echov "  - skipping KubeFlex Control Plane \"$name\""
+            continue
+        fi
```

Comment:
`kflex` does not duplicate the hosting cluster context. Maybe this script should also avoid that.

Reply:
Can you helpe me understand what you would like to do?
Create an ITS context that point to the existing hosting cluster user/server?
Honestly I'm concerned that it will require significant code changes because the `get_kubeconfig` is assumed to return a fully working kubeconfig for the CP

Reply:
I think that consistency with kflex is important. To keep that, we could change the definition of what this script is supposed to do, so that --- like kflex --- it is not supposed to create a(nother) context for the hosting cluster.

Reply:
I find extremely annoying to have to customize all the examples saying use `--context its1` is its type != host, use `--context <host-cluster>` if its1 isof type host.

Reply:
The examples are concrete, no "if" involved.